app/assets/javascripts/Components/autotest_manager.jsx

```diff
@@ -366,6 +373,21 @@ class AutotestManager extends React.Component {
               plugins: [labelPlugin()], // Ensure id is applied to visible input
             }}
           />
+          <label className="inline_label" htmlFor="token_end_date">
+            {I18n.t("activerecord.attributes.assignment.token_end_date")}
+          </label>
+          <Flatpickr
+            id="token_end_date"
+            value={this.state.token_end_date}
+            onChange={this.handleTokenEndDateChange}
+            options={{
```

Comment:
This is good; please also add *placeholder text* saying "Leave blank to use assignment due date". Remember to internationalize the string.